src/pages/protected/programs/singleProgram/singleProgram.page.tsx

```diff
@@ -330,36 +341,37 @@ export class SingleProgramPage extends React.Component<Props, State> {
           startTime={program?.programTimes.start_time}
           onClose={() => this.toggleEditTimeModal()}
         />
+        <WrappedEditImageModal
+          programId={this.props.match.params.id}
+          isOpen={this.state.openModal === SingeProgramPageModals.EDIT_IMAGE}
+          onCancel={() => this.setState({ openModal: null })}
+          onImageUpdated={() => {
+            this.setState({ openModal: null });
+            this.fetchProgram();
+          }}
+        />
       </Page>
     );
   }
-
-  private renderRecordedShowContent() {
-    const { program } = this.state;
-
-    if (program?.recorded_shows.length) {
-      return program?.recorded_shows.map((show) => {
-        return (
-          <tr key={show.id}>
-            <th>{show.id}</th>
-            <td>{show.name}</td>
-            <td>{show.duration}</td>
-            <td>{show.url}</td>
-            <td>{show.is_displayed ? "Yes" : "No"}</td>
-          </tr>
-        );
-      });
-    }
-
-    return <NoRecordedShows />;
-  }
-
-  private renderRecordedShowsTable() {
-    return (
-      <Table
-        columns={columns}
-        dataSource={this.state.program?.recorded_shows}
-      />
-    );
-  }
 }
+
+const WrappedEditImageModal: React.FC<{
```

Comment:
We are trying to deprecate MobX & the old API services & move to a hook & react-query based server state management.

To prevent a big refactor of the page to a Functional Component, I've wrapped the edit modal to use the hooks within, and after the update, to refetch the image